src/main/java/com/google/sps/servlets/Query.java

```diff
@@ -59,18 +113,83 @@ public void doPost(HttpServletRequest servletRequest, HttpServletResponse servle
+>>>>>>> resultsPage
 
         URL url = new URL(urlStr);
         URLConnection requestURL = url.openConnection();
         requestURL.connect();
 
         JsonParser jp = new JsonParser();
         JsonElement jsonElement = jp.parse(new InputStreamReader((InputStream) requestURL.getContent()));
+<<<<<<< HEAD
+        JsonObject jsonObj = jsonElement.getAsJsonObject();
+        System.out.println("json " + jsonObj.get("results"));
+        return jsonObj.get("results");
+    }
+
+    private JsonObject chooseRestaurant(JsonArray restaurants, Double price, Double rating){
+        Map<String, Integer> restaurantScores = new HashMap<>();
+        Collection<JsonObject> restaurantCollection = new Gson().fromJson(restaurants, new TypeToken<Collection<JsonObject>>(){}.getType());
+        int score;
+        int total = 0;
+
+        //Weights for how closely the searches follow the parameters
+        for(JsonObject restaurant : restaurantCollection){
+            score = 1;
+            try{
+                if(Double.parseDouble(restaurant.get("price_level").getAsString()) == price || price == 0)
+                    score += 10;
+                else if(Math.abs(Double.parseDouble(restaurant.get("price_level").getAsString())-price) <= 1)
+                    score += 6;
+                else if(Math.abs(Double.parseDouble(restaurant.get("price_level").getAsString())-price) <= 2)
+                    score += 3;
+                else if(Math.abs(Double.parseDouble(restaurant.get("price_level").getAsString())-price) <= 3)
+                    score += 1;
+            
+            } catch (Exception e){
+                System.out.println(e.getMessage());
+            }
+            try{
+                if(Double.parseDouble(restaurant.get("rating").getAsString()) == rating || rating == 0)
+                    score += 10;
+                else if(Math.abs(Double.parseDouble(restaurant.get("rating").getAsString())-rating) <= .5)
+                    score += 6;
+                else if(Math.abs(Double.parseDouble(restaurant.get("rating").getAsString())-rating) <= 1)
+                    score += 3;
+                else if(Math.abs(Double.parseDouble(restaurant.get("rating").getAsString())-rating) <= 1.5)
+                    score += 1;
+            } catch (Exception e){
+                System.out.println(e.getMessage());
+            }
+            total += score;
+            restaurantScores.put(restaurant.get("name").getAsString(), score);
+        }
+
+        //Generate a random number and parse through the scores to find
+        //the matching range
+        Random rd = new Random();
+        int selectedNum = rd.nextInt(total);
+        int value = 0;
+        String ans ="none";
+        for (String key : restaurantScores.keySet()) {
+            value += restaurantScores.get(key);
+            if(selectedNum <= value){
```

Comment:
this is sort of confusing.

why do you accumulate the restaurant scores into value and then pick the next restaurant that, cumulatively, brings the value higher than the randomly selected score?